database/sqlite3/sqlite3.go

```diff
@@ -96,12 +95,13 @@ func (m *Sqlite) Open(url string) (database.Driver, error) {
 	if err != nil {
 		return nil, err
 	}
-	dbfile := strings.Replace(migrate.FilterCustomQuery(purl).String(), "sqlite3://", "", 1)
+	dbfile := purl.Path
```

Comment:
We can't use `Path` directly since [options](https://github.com/mattn/go-sqlite3#dsn-examples) need to be passed to the underlying sqlite db driver

Reply:
That explains why that was using .String() in the first place...! 

So it looks like adding in `+ migrate.FilterCustomQuery(purl).Query().Encode()` should have the same net effect without escaping the path portion of the URL.   I've added the change and rebased to squash the commits into one.

PS. Thanks for the speedy review!  I didn't expect to see any movement on this today!

Reply:
Nope, I misread .Encode(), will need some logic to ensure it is either "?" + encodedQuery, or nothing.  Making another revision right now.